xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/search/searchSuggest.js

```diff
@@ -85,10 +86,11 @@ var XWiki = (function (XWiki) {
     onSuggestUpdated: function(event) {
       // Check if there are any suggestions, taking into account that there is at least one suggestion used to link the
       // search page.
-      if (event.memo.container.select('.suggestItem').length == 1) {
+      if (event.memo.container.select('.suggestItem').length === 1) {
         // Show the "No results!" message.
         this.noResultsMessage.removeClassName('hidden').setStyle({'float': 'left'});
       }
+      event.memo.container.select('.suggestItem')[0].href=event.memo.url||"";
```

Comment:
```suggestion
      event.memo.container.select('.suggestItem')[0].href = event.memo.url || "";
```

Also, this deserves a comment because it's not obvious why you're changing the URL of the first suggest item that is supposed to point to the search page.